schema/query/predicate.go

```diff
@@ -255,7 +271,23 @@ type NotEqual struct {
 
 // Match implements Expression interface.
 func (e NotEqual) Match(payload map[string]interface{}) bool {
-	return !reflect.DeepEqual(getField(payload, e.Field), e.Value)
+	value := getField(payload, e.Field)
+
+	// The $ne operator in MongoDB allows matching both single values and an
+	// array of values.
```

Comment:
FYI: Just did a small test in the MongoDB shell to confirm this is how `$ne` works.

```
> db.c.insert({"a":["a","b"]})
WriteResult({ "nInserted" : 1 })
> db.c.insert({"a":["c","b"]})
WriteResult({ "nInserted" : 1 })
> db.c.insert({"b":"z"})
WriteResult({ "nInserted" : 1 })
> db.c.find({"a":"a"})
{ "_id" : ObjectId("5bfc84458014a3148be6f734"), "a" : [ "a", "b" ] }
> db.c.find({"a":{"$ne":"a"}})
{ "_id" : ObjectId("5bfc844f8014a3148be6f735"), "a" : [ "c", "b" ] }
{ "_id" : ObjectId("5bfc846e8014a3148be6f736"), "b" : "z" }
```